### PR TITLE
Add support for Tinker Record (bsc#1086526)

### PR DESCRIPTION
**Pull request by jreidinger**
manual testing needed.

### CODE REVIEW

File: src/lib/cfa/ntp_conf.rb [hunk 1]
```diff
@@ -209,7 +210,9 @@ def reset_cache
       def record_entries
         return @record_entries if @record_entries
         matcher = Matcher.new do |k, _v|
-          RECORD_ENTRIES.include?(k.gsub("[]", ""))
```
Comment (teclator) [inline src/lib/cfa/ntp_conf.rb]: But that would include also not supported keys.. of course CFA will complain. Just thinking out loud..

File: src/lib/cfa/ntp_conf.rb [hunk 2]
```diff
@@ -239,8 +243,22 @@ def self.new_from_augeas(augeas_entry)
       # @param [string] key
       def self.record_class(key)
         entry_type = key.gsub("[]", "")
-        record_class = "::CFA::NtpConf::#{entry_type.capitalize}Record"
-        Kernel.const_get(record_class)
+        record_class = "#{entry_type.capitalize}Record"
+        if CFA::NtpConf.constants.include?(record_class.to_sym)
+          CFA::NtpConf.const_get(record_class.to_sym)
+        else
+          # it is not predefined record type, so use generic misc record
```
Comment (teclator) [inline src/lib/cfa/ntp_conf.rb]: NP: what about *it is not `a` predefined record type, so `a` generic misc record is used*

File: src/lib/cfa/ntp_conf.rb (hunk 1 above)
Comment (jreidinger) [reply src/lib/cfa/ntp_conf.rb]: we now support any key that appear. I am not sure how to do it when we have bunch of implemented keys.

Comment (teclator) [reply src/lib/cfa/ntp_conf.rb]: Just expected only the misc commands for using the `MiscRecord` and maybe a `GenericRecord` for the rest not defined yet.

File: test/ntp_client_test.rb [hunk 3]
```diff
@@ -89,6 +90,10 @@
         it "reads sync intervall" do
           expect(subject.sync_interval).to eq 15
         end
+
+        it "do not crash during write" do
```
Comment (teclator) [inline test/ntp_client_test.rb]: does

File: src/lib/cfa/ntp_conf.rb [hunk 4]
```diff
@@ -239,8 +243,22 @@ def self.new_from_augeas(augeas_entry)
...
-        record_class = "::CFA::NtpConf::#{entry_type.capitalize}Record"
-        Kernel.const_get(record_class)
+        record_class = "#{entry_type.capitalize}Record"
+        if CFA::NtpConf.constants.include?(record_class.to_sym)
+          CFA::NtpConf.const_get(record_class.to_sym)
+        else
+          # it is not a predefined record type, so use a generic misc record
+          res = Class.new(MiscRecord)
+          # and specify there its augeas type
+          res.define_method(:augeas_type) { entry_type + "[]" }
```
Comment (teclator) [inline src/lib/cfa/ntp_conf.rb]: defined_singleton_method ?

Comment (jreidinger) [reply src/lib/cfa/ntp_conf.rb]: no, it is not singleton method, it is instance method. But maybe something that can be changed and have it as class method and not instance one.

File: test/ntp_client_test.rb [hunk 5]
```diff
@@ -89,6 +90,13 @@
         it "reads sync intervall" do
           expect(subject.sync_interval).to eq 15
         end
+
+        it "following write succeed" do
```
Comment (teclator) [inline test/ntp_client_test.rb]: NP: it is not clear, at least for me.

**Comment by teclator** [comment]
Just for record purposes, I have tested it manually cloning the current config and writing it back with the expected results.

**Pull request merged**